ros2cli/ros2cli/node/strategy.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ros2cli.node.daemon import add_arguments as add_daemon_node_arguments
+from ros2cli.node.daemon import DaemonNode
+from ros2cli.node.daemon import is_daemon_running
+from ros2cli.node.daemon import spawn_daemon
+from ros2cli.node.direct import add_arguments as add_direct_node_arguments
+from ros2cli.node.direct import DirectNode
+
+
+class NodeStrategy(object):
+
+    def __init__(self, args):
+        if is_daemon_running(args):
+            self.node = DaemonNode(args)
+        else:
+            spawn_daemon(args)
+            self.node = DirectNode(args)
```

Comment:
In the case where the daemon is not running yet using the spawned daemon doesn't provide any benefit. Using the direct node is at least as fast as using the just spawned daemon and it allows the user to control e.g. the timeouts using command line arguments.